internal/logs/traces/discovery.go

```diff
@@ -0,0 +1,62 @@
+package traces
+
+import (
+	"time"
+
+	"github.com/ydb-platform/ydb-go-sdk/v3/internal/meta"
+	"github.com/ydb-platform/ydb-go-sdk/v3/logs"
+	"github.com/ydb-platform/ydb-go-sdk/v3/trace"
+)
+
+// Discovery makes trace.Discovery with logging events from details
+func Discovery(l logs.Logger, details trace.Details) (t trace.Discovery) {
+	if details&trace.DiscoveryEvents == 0 {
+		return
+	}
+	scope := []string{"discovery"}
+	ll := logger{
+		l:     l,
+		scope: scope,
+	}
+	t.OnDiscover = func(info trace.DiscoveryDiscoverStartInfo) func(trace.DiscoveryDiscoverDoneInfo) {
+		ll.Info("discover start",
+			logs.String("address", info.Address),
+			logs.String("database", info.Database),
+		)
+		start := time.Now()
+		return func(info trace.DiscoveryDiscoverDoneInfo) {
+			if info.Error == nil {
+				ll.Debug("discover done",
+					logs.Duration("latency", time.Since(start)),
```

Comment:
Several fields (latency, version) are very common between all traces. Maybe we should add something like
```go
func latency(start) logs.Field { return logs.Duration("latency", time.Since(start)) }

func version() logs.Field { return logs.String("version", meta.Version) }
```
in this package to DRY? The drawback of this is implicit behavior, which is not a big problem here IMO.

Reply:
about "version":

I think it can be common func-helper with "auto-add" version field, use pool for create fields slices, may be select log level, similar to [logDebugInfo](https://github.com/ydb-platform/ydb-go-sdk-zap/blob/35ffa3a654f6a8bffb6b31901fa8f60be2f0eece/loglevel.go#L12-L12) helper in current zap adapter (used for topic).

@asmyasnikov what do you think about the helper?

Reply:
Seems like this functionality could be added to [logger](https://github.com/xopoww/ydb-go-sdk/blob/364-logs/internal/logs/traces/logger.go) from `internal/logs/traces`.